### PR TITLE
Add -Distinct combinator

### PR DESCRIPTION
**Pull request by Sonichka1311**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add -Distinct combinator for aggregate functions.

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionDistinct.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        return nested_func->alignOfData();
+    }
+
+    bool hasTrivialDestructor() const override {
+        return nested_func->hasTrivialDestructor();
+    }
+
+    void add(AggregateDataPtr place, const IColumn ** columns, size_t row_num, Arena * arena) const override {
+        SipHash hash;
+        columns[0]->updateHashWithValue(row_num, hash);
```
Comment (CurtizJ) [inline src/AggregateFunctions/AggregateFunctionDistinct.h:86]: Functions with several arguments also can be supported. E.g. `corr(x, y)`.

**Review by CurtizJ — CHANGES_REQUESTED** [review]
Please, provide some tests.

File: src/AggregateFunctions/AggregateFunctionDistinct.h [hunk 2]
```diff
@@ -117,7 +117,7 @@ class AggregateFunctionDistinct final : public IAggregateFunctionHelper<Aggregat
         nested_func->deserialize(place, buf, arena);
     }
 
-    void insertResultInto(ConstAggregateDataPtr place, IColumn & to) const override
+    void insertResultInto(AggregateDataPtr place, IColumn & to) const override
```
Comment (Sonichka1311) [inline src/AggregateFunctions/AggregateFunctionDistinct.h:120]: Why? All aggregate functions uses `ConstAggregateDataPtr` as it declared in `IAggregateFunction`

**Comment by alexey-milovidov** [comment]
Discussed one issue in TG chat.

**Pull request merged**